### ISSUE TITLE
Locking feature doesn't unlock when the block is destroyed

### ISSUE THREAD

**Issue by Molitvan** [minecraft-access/minecraft-access#381]
When the locked on block is destroyed, the player is still locked on to that block's position until they unlock manually.

**Comment by lattaai5** [comment]
Hello, I tried to fix the issue

This is what I did:

Modified the LockingHandler class to check if the locked block still exists and unlock if it doesn't. This fixes the issue where the player remains locked onto a destroyed block's position.

I also created pull request: https://github.com/minecraft-access/minecraft-access/pull/382


> [!CAUTION]
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!
>
> The fix provided by Latta AI might not be complete and it can serve as an inspiration.

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt fixing issues on your repository, you can block this account.

**Comment by boholder** [comment]
There is auto unlock after destroying the block, but I noticed that if you press lock again, that position will be locked on. So the fixing method is remove the block and entity from candidate list when auto unlock from something.

**Comment by Molitvan** [comment]
> There is auto unlock after destroying the block, but I noticed that if you press lock again, that position will be locked on. So the fixing method is remove the block and entity from candidate list when auto unlock from something.

Not exactly. The destroyed block not being removed from the candidate list is definitely a separate issue I honestly didn't notice until now, but the current method for detecting if the target block has been destroyed doesn't work for most blocks. When you try it with something like a chest, it'll work, but something like an iron ore won't work. I'll fix it myself later today or tomorrow.
